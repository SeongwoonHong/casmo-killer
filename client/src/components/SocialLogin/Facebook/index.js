import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { loadSdk } from '@sharedUtils/socialAuth';
import LoadingCircle from '@sharedComponents/LoadingCircle';

class FacebookAuth extends Component {

  constructor(props) {

    super(props);

    this.state = {
      isSdkLoaded: false,
      isProcessing: false
    };

  }

  componentDidMount() {

    this._isMounted = true;

    if (document.getElementById('facebook-jssdk')) {
      this.sdkLoaded();
      return;
    }

    this.loadSdkLibrary();

  }

  componentWillUnmount() {

    this._isMounted = false;

  }

  setStateIfMounted(state) {

    if (this._isMounted) {
      this.setState(state);
    }

  }

  loadSdkLibrary() {

    const { clientId, version } = this.props;

    window.fbAsyncInit = () => {

      window.FB.init({
        version: `v${version}`,
        appId: clientId,
      });

      this.setStateIfMounted({
        isSdkLoaded: true
      });

    };

    loadSdk('facebook');

  }

  sdkLoaded() {

    this.setState({
      isSdkLoaded: true
    });

  }

  responseApi = (authResponse) => {

    const {
      // language: locale,
      // fields,
      onSuccess
    } = this.props;

    onSuccess({
      provider: 'facebook',
      accessToken: authResponse.accessToken
    });

    // window.FB.api('/me', {
    //   locale,
    //   fields
    // }, (me) => {
    //   Object.assign(me, authResponse);
    // });

  };

  checkLoginState = (response) => {

    const { authResponse, status } = response;
    const { onFailure } = this.props;

    this.setStateIfMounted({
      isProcessing: false
    });

    if (authResponse) {

      this.responseApi(authResponse);
      return;

    }

    if (onFailure) {
      onFailure({ status });
    }

  };

  click = () => {

    const { isSdkLoaded, isProcessing } = this.state;
    const { isDisabled, scope } = this.props;

    if (!isSdkLoaded || isProcessing || isDisabled) {
      return;
    }

    this.setState({
      isProcessing: true
    });

    window.FB.login(this.checkLoginState, { scope });

  };

  render() {

    const { children } = this.props;

    return (
      <button
        type="button"
        disabled={ !this.state.isSdkLoaded }
        onClick={ this.click }>
        {
          this.state.isSdkLoaded
            ? children
            : <LoadingCircle color="#515151" />
        }
      </button>
    );
  }

}


FacebookAuth.propTypes = {
  clientId: PropTypes.string.isRequired,
  onSuccess: PropTypes.func.isRequired,
  onFailure: PropTypes.func,
  scope: PropTypes.string,

  isDisabled: PropTypes.bool,
  version: PropTypes.string,
  children: PropTypes.node.isRequired,
};

FacebookAuth.defaultProps = {
  isDisabled: false,
  scope: 'public_profile,email',
  version: '2.9',
  onFailure: error => console.log(error)
};

export default FacebookAuth;
