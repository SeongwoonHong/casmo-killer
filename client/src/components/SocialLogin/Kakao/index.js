import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { loadSdk } from '@sharedUtils/socialAuth';
import LoadingCircle from '@sharedComponents/LoadingCircle';

class KakaoAuth extends Component {

  constructor(props) {

    super(props);

    this.state = {
      isSdkLoaded: false
    };

  }

  componentDidMount() {

    this.loadSdkLibrary();

  }

  async loadSdkLibrary() {

    const { clientId } = this.props;

    const shouldInit = await loadSdk('kakao');

    if (shouldInit) {
      window.Kakao.init(clientId);
    }

    this.setState({
      isSdkLoaded: true
    });

  }

  click = () => {

    const { onSuccess, onFailure } = this.props;

    window.Kakao.Auth.login({
      throughTalk: false,
      success: (response) => {
        onSuccess({
          provider: 'kakao',
          accessToken: response.access_token
        });
      },
      fail: onFailure
    });

  };

  render() {

    const { children } = this.props;

    return (
      <button
        type="button"
        disabled={ !this.state.isSdkLoaded }
        onClick={ this.click }>
        {
          this.state.isSdkLoaded
            ? children
            : <LoadingCircle />
        }
      </button>
    );

  }

}

KakaoAuth.propTypes = {
  clientId: PropTypes.string.isRequired,
  onSuccess: PropTypes.func.isRequired,
  onFailure: PropTypes.func,
  children: PropTypes.node.isRequired
};

KakaoAuth.defaultProps = {
  onFailure: error => console.log(error)
};

export default KakaoAuth;
