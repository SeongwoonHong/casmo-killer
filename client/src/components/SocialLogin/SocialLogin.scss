.Social-login {
    > .social-btn-wrapper {
        padding: $user-box-padding;
        > div {
            > button {
                @include flex-box();
                color: #fff;
                padding: 5px;
                width: 100%;
                > img {
                    @include rectanglify(30px);
                }
                > span {
                    font-size: 14px;
                    font-weight: 400;
                    width: 100%;
                }
                &:disabled {
                    > .preloader-wrapper {
                        @include rectanglify(30px);
                        margin: 0 auto;
                        > .spinner-layer {
                            border-color: palette(Grey, 900);
                        }
                    }
                }
            }
            &.facebook {
                background-color: $facebook-bg;
            }
            &.google {
                background-color: $google-bg;
            }
            &.kakao {
                background-color: $kakao-bg;
                > button {
                    color: $kakao-txt;
                }
            }
        }
    }
    @media screen and (min-width: 992px) {
        > .social-btn-wrapper {
            @include flex-box(space-between);
            > div {
                width: 33%;
            }
        }
    }
}
