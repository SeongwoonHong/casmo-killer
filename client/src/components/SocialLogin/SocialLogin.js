import React, { Component } from 'react';
import axios from 'axios';

import facebookImg from '@assets/facebook.png';
import googleImg from '@assets/google.png';
import kakaoImg from '@assets/kakao.png';

import FormMessage from '@sharedComponents/FormMessage';
import LoadingOverlay from '@sharedComponents/LoadingOverlay';

import './SocialLogin.scss';

import FacebookAuth from './Facebook';
import GoogleAuth from './Google';
import KakaoAuth from './Kakao';

class SocialLogin extends Component {

  constructor(props) {

    super(props);

    this.state = {
      isLoading: false,
      message: ''
    };

  }

  onSuccess = async (payload) => {

    const { onRegister, onSuccess } = this.props;

    this.setState({
      isLoading: true,
      message: ''
    });

    try {

      const { data } = await axios.post('/api/auth/login/social', payload);

      if (data) {

        if (data.shouldRegister) {
          onRegister(data.profile);
        } else if (data.user) {
          onSuccess(data.user);
        }

      } else {

        this.setState({
          isLoading: false,
          message: 'Failed to communicate with the server.'
        });

      }

    } catch (error) {

      console.error(error.response.data.error);
      this.setState({
        isLoading: false,
        message: error.response.data.message
      });

    }

  };

  render() {

    const {
      REACT_APP_facebookClientId: facebookId,
      REACT_APP_googleClientId: googleId,
      REACT_APP_kakaoClientId: kakaoId
    } = process.env;

    const { isLogin } = this.props;

    const { isLoading, message } = this.state;

    return (
      <div className="Social-login User__form__row">
        <LoadingOverlay
          isVisible={ isLoading }
          overlayColor="rgba(256,256,256,.75)"
          circleColor="#1F4B40" />
        <div className="User__form__header">
          <h3>Social Login</h3>
          {
            isLogin
              ? null
              : <p>User your favourite social network to sign up.</p>
          }
        </div>
        <FormMessage message={ message } />
        <div className="Social-login__row">
          <div className="Social-login__wrapper Social-login__wrapper--facebook">
            <FacebookAuth
              className="Social-login__btn"
              clientId={ facebookId }
              fields="name,email,picture"
              onSuccess={ this.onSuccess }>
              <img
                className="Social-login__btn__img"
                src={ facebookImg }
                alt="facebook-login" />
              <span className="Social-login__btn__text">
                Facebook
              </span>
            </FacebookAuth>
          </div>
          <div className="Social-login__wrapper Social-login__wrapper--google">
            <GoogleAuth
              className="Social-login__btn"
              clientId={ googleId }
              onSuccess={ this.onSuccess }>
              <img
                className="Social-login__btn__img"
                src={ googleImg }
                alt="google-login" />
              <span className="Social-login__btn__text">
                Google
              </span>
            </GoogleAuth>
          </div>
          <div className="Social-login__wrapper Social-login__wrapper--kakao">
            <KakaoAuth
              className="Social-login__btn"
              clientId={ kakaoId }
              onSuccess={ this.onSuccess }>
              <img
                className="Social-login__btn__img"
                src={ kakaoImg }
                alt="kakao-login" />
              <span className="Social-login__btn__text">
                Kakao
              </span>
            </KakaoAuth>
          </div>
        </div>
      </div>
    );
  }

}

export default SocialLogin;
