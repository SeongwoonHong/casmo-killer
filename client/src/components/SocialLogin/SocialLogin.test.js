import React from 'react';
import { shallow } from 'enzyme';

import SocialLogin from './SocialLogin';

// TODO:
// this component heavily relies on third party SDKs,
// so there is not much to test here.
describe('<SocialLogin />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(<SocialLogin />);
  });

  it('renders the component consistently', () => {
    expect(component).toMatchSnapshot();
  });

  it('should display correct message for login and registration', () => {

    expect(component.find('h3 + p').text()).toEqual('User your favourite social network to sign up.');

    component.setProps({ isLogin: true });

    expect(component.find('h3 + p').exists()).toEqual(false);

  });

});
