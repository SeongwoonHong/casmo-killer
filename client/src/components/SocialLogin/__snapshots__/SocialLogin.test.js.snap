// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLogin /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialLogin />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <LoadingOverlay
          circleColor="#1F4B40"
          circleSize=""
          isVisible={false}
          overlayColor="rgba(256,256,256,.75)"
/>,
        <div
          className="User__form__header"
>
          <h3>
                    Social Login
          </h3>
          <p>
                    User your favourite social network to sign up.
          </p>
</div>,
        <FormMessage
          className=""
          message=""
          type="error"
/>,
        <div
          className="Social-login__row"
>
          <div
                    className="Social-login__wrapper Social-login__wrapper--facebook"
          >
                    <FacebookAuth
                              className="Social-login__btn"
                              clientId="137341360260151"
                              fields="name,email,picture"
                              isDisabled={false}
                              onFailure={[Function]}
                              onSuccess={[Function]}
                              scope="public_profile,email"
                              version="2.9"
                    >
                              <img
                                        alt="facebook-login"
                                        className="Social-login__btn__img"
                                        src="facebook.png"
                              />
                              <span
                                        className="Social-login__btn__text"
                              >
                                        Facebook
                              </span>
                    </FacebookAuth>
          </div>
          <div
                    className="Social-login__wrapper Social-login__wrapper--google"
          >
                    <GoogleAuth
                              className="Social-login__btn"
                              clientId="754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com"
                              fetchBasicProfile={true}
                              hostedDomain=""
                              onFailure={[Function]}
                              onSuccess={[Function]}
                              prompt="select_account"
                              scope="profile email"
                              uxMode="popup"
                    >
                              <img
                                        alt="google-login"
                                        className="Social-login__btn__img"
                                        src="google.png"
                              />
                              <span
                                        className="Social-login__btn__text"
                              >
                                        Google
                              </span>
                    </GoogleAuth>
          </div>
          <div
                    className="Social-login__wrapper Social-login__wrapper--kakao"
          >
                    <KakaoAuth
                              className="Social-login__btn"
                              clientId="788acae6dc56555e6f4dc4064514177a"
                              onFailure={[Function]}
                              onSuccess={[Function]}
                    >
                              <img
                                        alt="kakao-login"
                                        className="Social-login__btn__img"
                                        src="kakao.png"
                              />
                              <span
                                        className="Social-login__btn__text"
                              >
                                        Kakao
                              </span>
                    </KakaoAuth>
          </div>
</div>,
      ],
      "className": "Social-login User__form__row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "circleColor": "#1F4B40",
          "circleSize": "",
          "isVisible": false,
          "overlayColor": "rgba(256,256,256,.75)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Social Login
</h3>,
            <p>
              User your favourite social network to sign up.
</p>,
          ],
          "className": "User__form__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Social Login",
            },
            "ref": null,
            "rendered": "Social Login",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "User your favourite social network to sign up.",
            },
            "ref": null,
            "rendered": "User your favourite social network to sign up.",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "message": "",
          "type": "error",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Social-login__wrapper Social-login__wrapper--facebook"
>
              <FacebookAuth
                            className="Social-login__btn"
                            clientId="137341360260151"
                            fields="name,email,picture"
                            isDisabled={false}
                            onFailure={[Function]}
                            onSuccess={[Function]}
                            scope="public_profile,email"
                            version="2.9"
              >
                            <img
                                          alt="facebook-login"
                                          className="Social-login__btn__img"
                                          src="facebook.png"
                            />
                            <span
                                          className="Social-login__btn__text"
                            >
                                          Facebook
                            </span>
              </FacebookAuth>
</div>,
            <div
              className="Social-login__wrapper Social-login__wrapper--google"
>
              <GoogleAuth
                            className="Social-login__btn"
                            clientId="754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com"
                            fetchBasicProfile={true}
                            hostedDomain=""
                            onFailure={[Function]}
                            onSuccess={[Function]}
                            prompt="select_account"
                            scope="profile email"
                            uxMode="popup"
              >
                            <img
                                          alt="google-login"
                                          className="Social-login__btn__img"
                                          src="google.png"
                            />
                            <span
                                          className="Social-login__btn__text"
                            >
                                          Google
                            </span>
              </GoogleAuth>
</div>,
            <div
              className="Social-login__wrapper Social-login__wrapper--kakao"
>
              <KakaoAuth
                            className="Social-login__btn"
                            clientId="788acae6dc56555e6f4dc4064514177a"
                            onFailure={[Function]}
                            onSuccess={[Function]}
              >
                            <img
                                          alt="kakao-login"
                                          className="Social-login__btn__img"
                                          src="kakao.png"
                            />
                            <span
                                          className="Social-login__btn__text"
                            >
                                          Kakao
                            </span>
              </KakaoAuth>
</div>,
          ],
          "className": "Social-login__row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FacebookAuth
                className="Social-login__btn"
                clientId="137341360260151"
                fields="name,email,picture"
                isDisabled={false}
                onFailure={[Function]}
                onSuccess={[Function]}
                scope="public_profile,email"
                version="2.9"
>
                <img
                                alt="facebook-login"
                                className="Social-login__btn__img"
                                src="facebook.png"
                />
                <span
                                className="Social-login__btn__text"
                >
                                Facebook
                </span>
</FacebookAuth>,
              "className": "Social-login__wrapper Social-login__wrapper--facebook",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt="facebook-login"
                    className="Social-login__btn__img"
                    src="facebook.png"
/>,
                  <span
                    className="Social-login__btn__text"
>
                    Facebook
</span>,
                ],
                "className": "Social-login__btn",
                "clientId": "137341360260151",
                "fields": "name,email,picture",
                "isDisabled": false,
                "onFailure": [Function],
                "onSuccess": [Function],
                "scope": "public_profile,email",
                "version": "2.9",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "facebook-login",
                    "className": "Social-login__btn__img",
                    "src": "facebook.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Facebook",
                    "className": "Social-login__btn__text",
                  },
                  "ref": null,
                  "rendered": "Facebook",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <GoogleAuth
                className="Social-login__btn"
                clientId="754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com"
                fetchBasicProfile={true}
                hostedDomain=""
                onFailure={[Function]}
                onSuccess={[Function]}
                prompt="select_account"
                scope="profile email"
                uxMode="popup"
>
                <img
                                alt="google-login"
                                className="Social-login__btn__img"
                                src="google.png"
                />
                <span
                                className="Social-login__btn__text"
                >
                                Google
                </span>
</GoogleAuth>,
              "className": "Social-login__wrapper Social-login__wrapper--google",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt="google-login"
                    className="Social-login__btn__img"
                    src="google.png"
/>,
                  <span
                    className="Social-login__btn__text"
>
                    Google
</span>,
                ],
                "className": "Social-login__btn",
                "clientId": "754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com",
                "fetchBasicProfile": true,
                "hostedDomain": "",
                "onFailure": [Function],
                "onSuccess": [Function],
                "prompt": "select_account",
                "scope": "profile email",
                "uxMode": "popup",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "google-login",
                    "className": "Social-login__btn__img",
                    "src": "google.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Google",
                    "className": "Social-login__btn__text",
                  },
                  "ref": null,
                  "rendered": "Google",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <KakaoAuth
                className="Social-login__btn"
                clientId="788acae6dc56555e6f4dc4064514177a"
                onFailure={[Function]}
                onSuccess={[Function]}
>
                <img
                                alt="kakao-login"
                                className="Social-login__btn__img"
                                src="kakao.png"
                />
                <span
                                className="Social-login__btn__text"
                >
                                Kakao
                </span>
</KakaoAuth>,
              "className": "Social-login__wrapper Social-login__wrapper--kakao",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt="kakao-login"
                    className="Social-login__btn__img"
                    src="kakao.png"
/>,
                  <span
                    className="Social-login__btn__text"
>
                    Kakao
</span>,
                ],
                "className": "Social-login__btn",
                "clientId": "788acae6dc56555e6f4dc4064514177a",
                "onFailure": [Function],
                "onSuccess": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "kakao-login",
                    "className": "Social-login__btn__img",
                    "src": "kakao.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kakao",
                    "className": "Social-login__btn__text",
                  },
                  "ref": null,
                  "rendered": "Kakao",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <LoadingOverlay
            circleColor="#1F4B40"
            circleSize=""
            isVisible={false}
            overlayColor="rgba(256,256,256,.75)"
/>,
          <div
            className="User__form__header"
>
            <h3>
                        Social Login
            </h3>
            <p>
                        User your favourite social network to sign up.
            </p>
</div>,
          <FormMessage
            className=""
            message=""
            type="error"
/>,
          <div
            className="Social-login__row"
>
            <div
                        className="Social-login__wrapper Social-login__wrapper--facebook"
            >
                        <FacebookAuth
                                    className="Social-login__btn"
                                    clientId="137341360260151"
                                    fields="name,email,picture"
                                    isDisabled={false}
                                    onFailure={[Function]}
                                    onSuccess={[Function]}
                                    scope="public_profile,email"
                                    version="2.9"
                        >
                                    <img
                                                alt="facebook-login"
                                                className="Social-login__btn__img"
                                                src="facebook.png"
                                    />
                                    <span
                                                className="Social-login__btn__text"
                                    >
                                                Facebook
                                    </span>
                        </FacebookAuth>
            </div>
            <div
                        className="Social-login__wrapper Social-login__wrapper--google"
            >
                        <GoogleAuth
                                    className="Social-login__btn"
                                    clientId="754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com"
                                    fetchBasicProfile={true}
                                    hostedDomain=""
                                    onFailure={[Function]}
                                    onSuccess={[Function]}
                                    prompt="select_account"
                                    scope="profile email"
                                    uxMode="popup"
                        >
                                    <img
                                                alt="google-login"
                                                className="Social-login__btn__img"
                                                src="google.png"
                                    />
                                    <span
                                                className="Social-login__btn__text"
                                    >
                                                Google
                                    </span>
                        </GoogleAuth>
            </div>
            <div
                        className="Social-login__wrapper Social-login__wrapper--kakao"
            >
                        <KakaoAuth
                                    className="Social-login__btn"
                                    clientId="788acae6dc56555e6f4dc4064514177a"
                                    onFailure={[Function]}
                                    onSuccess={[Function]}
                        >
                                    <img
                                                alt="kakao-login"
                                                className="Social-login__btn__img"
                                                src="kakao.png"
                                    />
                                    <span
                                                className="Social-login__btn__text"
                                    >
                                                Kakao
                                    </span>
                        </KakaoAuth>
            </div>
</div>,
        ],
        "className": "Social-login User__form__row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "circleColor": "#1F4B40",
            "circleSize": "",
            "isVisible": false,
            "overlayColor": "rgba(256,256,256,.75)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Social Login
</h3>,
              <p>
                User your favourite social network to sign up.
</p>,
            ],
            "className": "User__form__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Social Login",
              },
              "ref": null,
              "rendered": "Social Login",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "User your favourite social network to sign up.",
              },
              "ref": null,
              "rendered": "User your favourite social network to sign up.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "message": "",
            "type": "error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Social-login__wrapper Social-login__wrapper--facebook"
>
                <FacebookAuth
                                className="Social-login__btn"
                                clientId="137341360260151"
                                fields="name,email,picture"
                                isDisabled={false}
                                onFailure={[Function]}
                                onSuccess={[Function]}
                                scope="public_profile,email"
                                version="2.9"
                >
                                <img
                                                alt="facebook-login"
                                                className="Social-login__btn__img"
                                                src="facebook.png"
                                />
                                <span
                                                className="Social-login__btn__text"
                                >
                                                Facebook
                                </span>
                </FacebookAuth>
</div>,
              <div
                className="Social-login__wrapper Social-login__wrapper--google"
>
                <GoogleAuth
                                className="Social-login__btn"
                                clientId="754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com"
                                fetchBasicProfile={true}
                                hostedDomain=""
                                onFailure={[Function]}
                                onSuccess={[Function]}
                                prompt="select_account"
                                scope="profile email"
                                uxMode="popup"
                >
                                <img
                                                alt="google-login"
                                                className="Social-login__btn__img"
                                                src="google.png"
                                />
                                <span
                                                className="Social-login__btn__text"
                                >
                                                Google
                                </span>
                </GoogleAuth>
</div>,
              <div
                className="Social-login__wrapper Social-login__wrapper--kakao"
>
                <KakaoAuth
                                className="Social-login__btn"
                                clientId="788acae6dc56555e6f4dc4064514177a"
                                onFailure={[Function]}
                                onSuccess={[Function]}
                >
                                <img
                                                alt="kakao-login"
                                                className="Social-login__btn__img"
                                                src="kakao.png"
                                />
                                <span
                                                className="Social-login__btn__text"
                                >
                                                Kakao
                                </span>
                </KakaoAuth>
</div>,
            ],
            "className": "Social-login__row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FacebookAuth
                  className="Social-login__btn"
                  clientId="137341360260151"
                  fields="name,email,picture"
                  isDisabled={false}
                  onFailure={[Function]}
                  onSuccess={[Function]}
                  scope="public_profile,email"
                  version="2.9"
>
                  <img
                                    alt="facebook-login"
                                    className="Social-login__btn__img"
                                    src="facebook.png"
                  />
                  <span
                                    className="Social-login__btn__text"
                  >
                                    Facebook
                  </span>
</FacebookAuth>,
                "className": "Social-login__wrapper Social-login__wrapper--facebook",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <img
                      alt="facebook-login"
                      className="Social-login__btn__img"
                      src="facebook.png"
/>,
                    <span
                      className="Social-login__btn__text"
>
                      Facebook
</span>,
                  ],
                  "className": "Social-login__btn",
                  "clientId": "137341360260151",
                  "fields": "name,email,picture",
                  "isDisabled": false,
                  "onFailure": [Function],
                  "onSuccess": [Function],
                  "scope": "public_profile,email",
                  "version": "2.9",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "facebook-login",
                      "className": "Social-login__btn__img",
                      "src": "facebook.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Facebook",
                      "className": "Social-login__btn__text",
                    },
                    "ref": null,
                    "rendered": "Facebook",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <GoogleAuth
                  className="Social-login__btn"
                  clientId="754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com"
                  fetchBasicProfile={true}
                  hostedDomain=""
                  onFailure={[Function]}
                  onSuccess={[Function]}
                  prompt="select_account"
                  scope="profile email"
                  uxMode="popup"
>
                  <img
                                    alt="google-login"
                                    className="Social-login__btn__img"
                                    src="google.png"
                  />
                  <span
                                    className="Social-login__btn__text"
                  >
                                    Google
                  </span>
</GoogleAuth>,
                "className": "Social-login__wrapper Social-login__wrapper--google",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <img
                      alt="google-login"
                      className="Social-login__btn__img"
                      src="google.png"
/>,
                    <span
                      className="Social-login__btn__text"
>
                      Google
</span>,
                  ],
                  "className": "Social-login__btn",
                  "clientId": "754520959165-qfcdlss8glpjrjs59oldmtnj24m7tfk6.apps.googleusercontent.com",
                  "fetchBasicProfile": true,
                  "hostedDomain": "",
                  "onFailure": [Function],
                  "onSuccess": [Function],
                  "prompt": "select_account",
                  "scope": "profile email",
                  "uxMode": "popup",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "google-login",
                      "className": "Social-login__btn__img",
                      "src": "google.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Google",
                      "className": "Social-login__btn__text",
                    },
                    "ref": null,
                    "rendered": "Google",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <KakaoAuth
                  className="Social-login__btn"
                  clientId="788acae6dc56555e6f4dc4064514177a"
                  onFailure={[Function]}
                  onSuccess={[Function]}
>
                  <img
                                    alt="kakao-login"
                                    className="Social-login__btn__img"
                                    src="kakao.png"
                  />
                  <span
                                    className="Social-login__btn__text"
                  >
                                    Kakao
                  </span>
</KakaoAuth>,
                "className": "Social-login__wrapper Social-login__wrapper--kakao",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <img
                      alt="kakao-login"
                      className="Social-login__btn__img"
                      src="kakao.png"
/>,
                    <span
                      className="Social-login__btn__text"
>
                      Kakao
</span>,
                  ],
                  "className": "Social-login__btn",
                  "clientId": "788acae6dc56555e6f4dc4064514177a",
                  "onFailure": [Function],
                  "onSuccess": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "kakao-login",
                      "className": "Social-login__btn__img",
                      "src": "kakao.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kakao",
                      "className": "Social-login__btn__text",
                    },
                    "ref": null,
                    "rendered": "Kakao",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
